### PR TITLE
subtests to skip on jetson CI

### PR DESCRIPTION
**Pull request by puririshi98**
In order to get  a successful self test run on jetsons we need these skipifs and then run the following command:

`python3 test/run_test.py --continue-through-error -x distributed/_sharded_tensor/ops/test_embedding distributed/_sharded_tensor/ops/test_embedding_bag distributed/_sharded_tensor/ops/test_init distributed/_sharded_tensor/ops/test_linear distributed/_sharded_tensor/test_sharded_tensor distributed/_sharding_spec/test_sharding_spec distributed/algorithms/test_join distributed/elastic/events/lib_test distributed/elastic/metrics/api_test distributed/elastic/multiprocessing/api_test distributed/elastic/timer/api_test distributed/elastic/timer/local_timer_example distributed/elastic/timer/local_timer_test distributed/elastic/utils/distributed_test distributed/elastic/utils/logging_test distributed/elastic/utils/util_test distributed/fsdp/test_flatten_params_wrapper distributed/fsdp/test_fsdp_core distributed/fsdp/test_fsdp_freezing_weights distributed/fsdp/test_fsdp_input distributed/fsdp/test_fsdp_memory distributed/fsdp/test_fsdp_multiple_forward distributed/fsdp/test_fsdp_multiple_wrapping distributed/fsdp/test_fsdp_overlap distributed/fsdp/test_fsdp_param_mutation distributed/fsdp/test_fsdp_pure_fp16 distributed/fsdp/test_fsdp_uneven distributed/fsdp/test_utils distributed/fsdp/test_wrap distributed/nn/jit/test_instantiator distributed/optim/test_zero_redundancy_optimizer distributed/pipeline/sync/skip/test_api distributed/pipeline/sync/skip/test_gpipe distributed/pipeline/sync/skip/test_inspect_skip_layout distributed/pipeline/sync/skip/test_leak distributed/pipeline/sync/skip/test_portal distributed/pipeline/sync/skip/test_stash_pop distributed/pipeline/sync/skip/test_tracker distributed/pipeline/sync/skip/test_verify_skippables distributed/pipeline/sync/test_balance distributed/pipeline/sync/test_bugs distributed/pipeline/sync/test_checkpoint distributed/pipeline/sync/test_copy distributed/pipeline/sync/test_deferred_batch_norm distributed/pipeline/sync/test_dependency distributed/pipeline/sync/test_inplace distributed/pipeline/sync/test_microbatch distributed/pipeline/sync/test_phony distributed/pipeline/sync/test_pipe distributed/pipeline/sync/test_pipeline distributed/pipeline/sync/test_stream distributed/pipeline/sync/test_transparency distributed/pipeline/sync/test_worker distributed/rpc/cuda/test_tensorpipe_agent distributed/rpc/test_faulty_agent distributed/rpc/test_tensorpipe_agent distributed/test_c10d_common distributed/test_c10d_gloo distributed/test_c10d_nccl distributed/test_c10d_spawn_gloo distributed/test_c10d_spawn_nccl distributed/test_data_parallel distributed/test_distributed_spawn distributed/test_launcher distributed/test_nccl distributed/test_pg_wrapper distributed/test_store test_binary_ufuncs test_autograd test_ao_sparsity distributions/test_constraints distributions/test_distributions test_cpp_extensions_aot_ninja test_cpp_extensions_aot_no_ninja test_cuda_primary_ctx test_functional_optim test_fx test_fx_experimental test_jit test_jit_cuda_fuser test_jit_legacy test_jit_profiling test_linalg test_multiprocessing test_namedtuple_return_api test_ops test_quantization test_sparse_csr test_tensorexpr test_nn test_serialization -- -v --save-xml; python3 test/test_nn.py -v; python3 test/test_serialization.py -v`

### CODE REVIEW

File: test/test_cuda.py [hunk 1]
```diff
@@ -274,6 +275,7 @@ def assert_change(comp=1, empty_cache=False, reset_peak=False):
         assert_change(0, empty_cache=True)
         assert_change(0, reset_peak=True)
 
+    @unittest.skipIf(IS_JETSON, "Does not work on jetson")
```
Comment (ptrblck) [inline test/test_cuda.py:278]: Do you know what errors are raised on Jetsons for this test?

File: test/test_indexing.py [hunk 2]
```diff
@@ -727,6 +729,7 @@ def test_byte_mask_accumulate(self, device):
             self.assertEqual(y, torch.ones(size=(10, 10), device=device))
             self.assertEquals(len(w), 2)
 
+    @unittest.skipIf(IS_JETSON, "Too large for Jetson")
```
Comment (ptrblck) [inline test/test_indexing.py:732]: I assume this test runs OOM? If so, would it make more sense to check the memory requirements via e.g. https://github.com/pytorch/pytorch/blob/67941c8a94e1731c1f33f37915d210ec244be589/test/test_cuda.py#L57-L58

File: test/test_module_init.py [hunk 3]
```diff
@@ -231,6 +219,15 @@ def build_constructor_arg_db():
...
         torch.nn.quantized.QFunctional: ((), {}),
     }
+    if IS_JETSON:
+        x[torch.nn.qat.Conv2d] = ((3, 3, 3), {'qconfig': torch.ao.quantization.default_qconfig})
+        x[torch.nn.qat.Conv3d] = ((3, 3, 3), {'qconfig': torch.ao.quantization.default_qconfig})
+        x[torch.nn.qat.Linear] = ((5, 2), {'qconfig': torch.ao.quantization.default_qconfig,})
+        x[torch.nn.qat.Embedding] =
+            ((10, 12), {'qconfig': torch.ao.quantization.float_qparams_weight_only_qconfig})
+        x[torch.nn.qat.EmbeddingBag] =
+            ((10, 12), {'qconfig': torch.ao.quantization.float_qparams_weight_only_qconfig})
```
Comment (ptrblck) [inline test/test_module_init.py:229]: Are these tests now only executed on Jetson and skipped otherwise?

File: test/test_nn.py [hunk 4]
```diff
@@ -14142,6 +14150,8 @@ def test_TransformerDecoderLayer_empty(self, device):
     @expectedFailureMeta  # RuntimeError: cannot reshape tensor of 0 elements into shape [1, 0, -1]
     @onlyNativeDeviceTypes
     def test_TransformerDecoder_empty(self, device):
+        if device != 'cpu' and IS_JETSON:
+            self.skipTest('Killed on Jetson GPU')
```
Comment (ptrblck) [inline test/test_nn.py:14154]: Is this test running OOM?

File: test/test_indexing.py (hunk 2 above)
Comment (puririshi98) [reply test/test_indexing.py:732]: okay will do

File: test/test_module_init.py (hunk 3 above)
Comment (puririshi98) [reply test/test_module_init.py:229]: actually intended to be `not IS_JETSON` which I could have sworn was necessary but upon rechecking it seems these tests are skipped automatically and the changes to this file are not needed.

File: test/test_nn.py (hunk 4 above)
Comment (puririshi98) [reply test/test_nn.py:14154]: it is a flaky test where sometimes on Jetson the test just stops with no message other than "Killed"

File: test/test_cuda.py (hunk 1 above)
Comment (puririshi98) [reply test/test_cuda.py:278]:
```
FAIL: test_cudart_register (__main__.TestCuda)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/nvidia/.local/lib/python3.6/site-packages/torch/testing/_internal/common_utils.py", line 1470, in wrapper
    method(*args, **kwargs)
  File "/home/nvidia/.local/lib/python3.6/site-packages/torch/testing/_internal/common_utils.py", line 1470, in wrapper
    method(*args, **kwargs)
  File "test_cuda.py", line 284, in test_cudart_register
    self.assertEqual(r, 0)
  File "/home/nvidia/.local/lib/python3.6/site-packages/torch/testing/_internal/common_utils.py", line 2074, in assertEqual
    super().assertEqual(x, y, msg=msg)
AssertionError: <cudaError.???: 801> != 0

----------------------------------------------------------------------
Ran 168 tests in 42.730s
```

**Comment by mcarilli** [comment]
Is this skipping some test files wholesale? We shouldn't do that unless we have no other choice. What are the typical failure patterns you see for the failing tests/files?

**Comment by puririshi98** [comment]
> Is this skipping some test files wholesale? We shouldn't do that unless we have no other choice. What are the typical failure patterns you see for the failing tests/files?

The changes being PRd only skip some subtests that fail by checking if the device running it is a Jetson. When I actually run the full test suite on the jetson there are some full test files that fail on every subtest or almost every subtest due to incompatibility issues

**Pull request closed without merging**